Library/Homebrew/github_packages.rb

```diff
@@ -241,8 +241,21 @@ def upload_bottle(user, token, skopeo, formula_full_name, bottle_hash, dry_run:)
         MacOS::Version.from_symbol(bottle_tag).to_s
       end
 
+      variant = if architecture == "arm64"
+        "v8"
+      elsif tab["oldest_cpu_family"]
+        tab["oldest_cpu_family"]
+      elsif architecture == "amd64"
+        if os == "darwin"
+          Hardware.oldest_cpu(OS::Mac::Version.new(os_version[/macOS ([0-9]+\.[0-9]+)/, 1])).to_s
+        else
+          "core2"
+        end
+      end
+
       platform_hash = {
         architecture: architecture,
+        variant: variant,
```

Comment:
I don't think we should set this on macOS. It's displayed in the GitHub Packages interface and we've asked for the to display OS version. It doesn't make sense to display both on macOS where the OS version implies the oldest CPU.

Reply:
>> I  don't think we should set this on macOS.

> Some software benefits greatly in performance by many fold when able to use modern SIMD instructions. I'd like to and plan to implement building bottles for newer CPU architectures on Linux.

This same argument applies to macOS as it does on Linux. A third-party tap like Brewsci/bio could provide a bottle for the minimum version supported by that version of macOS, and a second bottle that targets a modern SIMD architecture like AVX-512.

Reply:
We don't need to make use of this feature for Homebrew/core if we don't want to, but it'd be great for the package manager itself to support this feature.

Reply:
It's possible but it's not something we're going to do on macOS for Homebrew/core so we shouldn't be setting this on macOS. This will make the UI for GitHub Packages on macOS worse today.

If this argument also applies on Linux: cool, then I suggest not adding it there either.

I'm trying to defer to you slightly more here on Linux and on metadata but I really don't want to see us bloating tabs or metadata on macOS for something we have no plans to do in future. 

Bintray is going away in less than a month. Let's revisit and reconsider these sort of things when GitHub Packages is actually working for our users and we've migrated everything away from Bintray.